symlink-vendor-directory.js

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable flowtype/require-valid-file-annotation */
+/* eslint-disable import/no-nodejs-modules */
+
+const {exec} = require('child_process'); //
+const path = require('path');
+
+// the sulu/skeleton uses the assets/admin directory as root directory for the administration interface javascript
+// application. this application imports the sulu/sulu code by requiring the bundles from the the vendor directory
+// in its package.json.
+//
+// unfortunately this setup causes various problems in the javascript ecosystem because the assets/admin directory
+// is not an ancestor of the vendor/sulu/sulu directory. for example, npm 7 does not install the dependencies of the
+// bundle packages (https://github.com/npm/cli/issues/2339) and older npm versions are not able to dedupe packages that
+// are required by a bundle and in the assets/admin directory.
+// because of this, packages that were not deduped might be included in the webpack build multiple times. this
+// increases the size of the build and will lead to an error in case of the @ckeditor packages:
+// https://ckeditor.com/docs/ckeditor5/latest/framework/guides/support/error-codes.html#error-ckeditor-duplicated-modules
+//
+// to prevent these errors, this file creates a assets/admin/node_modules/node_modules/@sulu/vendor symlink and use
+// the symlink to require the sulu bundles in the assets/admin/package.json. this makes the vendor directory a child
+// of the assets/admin directory and allows npm to correctly dedupe the installed packages.
```

Comment:
```suggestion
// the sulu/skeleton uses the assets/admin directory as root directory for the administration interface javascript
// application. this application imports the sulu/sulu code by requiring the bundles from the the vendor directory
// in its package.json.
//
// unfortunately this setup causes various problems in the javascript ecosystem because the assets/admin directory
// is not an ancestor of the vendor/sulu/sulu directory which is used to require the code of bundles. for example,
// npm 7 does not install the dependencies of the bundle packages (https://github.com/npm/cli/issues/2339) and older
// npm versions are not able to dedupe packages that are required by a bundle and in the assets/admin directory.
// because of this, packages that were not deduped might be included in the webpack build multiple times. this
// increases the size of the build and will lead to an error in case of the @ckeditor packages:
// https://ckeditor.com/docs/ckeditor5/latest/framework/guides/support/error-codes.html#error-ckeditor-duplicated-modules
//
// to prevent these problems, this file creates a assets/admin/node_modules/node_modules/@sulu/vendor symlink and we
// use the symlinked vendor directory to require the bundles in the assets/admin/package.json.
// this makes the directory that is used for requiring the bundles a descendant of the assets/admin directory and
// allows npm to correctly dedupe the installed packages.
```